PIL/Image.py

```diff
@@ -504,26 +530,30 @@ class Image(object):
     _close_exclusive_fp_after_loading = True
 
     def __init__(self):
+        # type: () -> None
         # FIXME: take "new" parameters / other image?
         # FIXME: turn mode and size into delegating properties?
-        self.im = None
+        self.im = None  # type: Optional[ImagingCore]
         self.mode = ""
         self.size = (0, 0)
-        self.palette = None
-        self.info = {}
+        self.palette = None  # type: Optional[ImagePalette.ImagePalette]
+        self.info = {}  # type: Dict[Text, Any]
         self.category = NORMAL
         self.readonly = 0
         self.pyaccess = None
 
     @property
     def width(self):
+        # type: () -> int
         return self.size[0]
 
     @property
     def height(self):
+        # type: () -> int
         return self.size[1]
 
     def _new(self, im):
+        # type: (Image) -> Image
```

Comment:
@wiredfool Regarding `ImagingCore`, what about [this type](https://github.com/python-pillow/Pillow/pull/2687#discussion-diff-134087659R1338)?